ilastik/utility/exportFile.py

```diff
@@ -249,35 +249,38 @@ def create_slicing(axistags, dimensions, margin, feature_table):
         yields also the actual object id
     """
     assert margin >= 0, "Margin muss be greater than or equal to 0"
-    time = feature_table[Default.TimeColumnName].astype(np.int32)
-    minx = feature_table["Bounding Box Minimum_0"].astype(np.int32)
-    maxx = feature_table["Bounding Box Maximum_0"].astype(np.int32)
-    miny = feature_table["Bounding Box Minimum_1"].astype(np.int32)
-    maxy = feature_table["Bounding Box Maximum_1"].astype(np.int32)
-    table_shape = feature_table.shape[0]
-    try:
-        minz = feature_table["Bounding Box Minimum_2"].astype(np.int32)
-        maxz = feature_table["Bounding Box Maximum_2"].astype(np.int32)
-    except ValueError:
-        minz = [0] * table_shape
-        maxz = [1] * table_shape
-
-    indices = list(map(axistags.index, "txyzc"))
-    excludes = indices.count(-1)
-    oid = 1
-    for i in range(table_shape):
-        if time[i] != time[i - 1]:
-            oid = 1
-        # noinspection PyTypeChecker
-        slicing = [
-            slice(time[i], time[i] + 1),
-            _slice_range(minx[i] - margin, maxx[i] + margin, dimensions[1]),
-            _slice_range(miny[i] - margin, maxy[i] + margin, dimensions[2]),
-            _slice_range(minz[i] - margin, maxz[i] + margin, dimensions[3]),
-            slice(None),
-        ]
-        yield [slicing[x] for x in indices][: 5 - excludes], oid
-        oid += 1
+    if len(feature_table) == 0:
+        yield from ()
```

Comment:
hmmm I had this first, and then thought for some reason that it would be more clear with the if else. But you're right, there's the danger of someone adding code after the currently indented one ...

Reply:
I suggested this mainly as an early exit pattern, in order to reduce indentation.

Reply:
Oh, and you also don't need `yield from ()` here: since the function already contains the `yield` keyword, it will be treated as a generator function.

Reply:
didn't know this automagic.